macros/evm/evm_live_views.sql

```diff
@@ -0,0 +1,1638 @@
+{% macro evm_live_view_latest_block_height(schema, blockchain, network) %}
+    SELECT
+        {{ schema }}.udf_rpc('eth_blockNumber', []) as result,
+        utils.udf_hex_to_int(result)::integer AS latest_block_height,
+        COALESCE(
+            block_height,
+            latest_block_height
+        ) AS min_height,
+        iff(
+            COALESCE(to_latest, false),
+            latest_block_height,
+            min_height
+        ) AS max_height
+{% endmacro %}
+
+{% macro evm_live_view_target_blocks(schema, blockchain, network, batch_size=10) %}
+    WITH heights AS (
+        {{ evm_live_view_latest_block_height(schema, blockchain, network) | indent(4) -}}
+    ),
+    block_spine AS (
+        SELECT
+            ROW_NUMBER() OVER (
+                ORDER BY
+                    NULL
+            ) - 1 + COALESCE(block_height, latest_block_height)::integer AS block_number,
+            min_height,
+            IFF(
+                COALESCE(to_latest, false),
+                block_height,
+                min_height
+            ) AS max_height,
+            latest_block_height
+        FROM
+            TABLE(generator(ROWCOUNT => 1000)),
```

Comment:
`ROWCOUNT`  is deterministic since we always know `max_height` and `min_height`. 
```suggestion
            TABLE(generator(ROWCOUNT => (max_height - min_height + 1))),
```

Reply:
This also feels like a premature optimization as the qualify below also filters for this range.